new-log-viewer/src/components/Editor/MonacoInstance/index.tsx

```diff
@@ -0,0 +1,127 @@
+import {
+    useEffect,
+    useRef,
+} from "react";
+
+import * as monaco from "monaco-editor/esm/vs/editor/editor.api";
+
+import {ActionType} from "../../../utils/actions";
+import {
+    CursorExplicitPosChangeCallback,
+    CustomActionCallback,
+} from "./typings";
+import {
+    goToPositionAndCenter,
+    initMonacoEditor,
+} from "./utils";
+
+import "./index.css";
+
+
+interface MonacoEditorProps {
+    lineNum: number,
+    text: string,
+    actions: ActionType[],
+    beforeMount?: () => void,
+    beforeTextUpdate?: (editor: monaco.editor.IStandaloneCodeEditor) => void,
+    onCursorExplicitPosChange: CursorExplicitPosChangeCallback,
+    onCustomAction: CustomActionCallback,
+    onMount?: (editor: monaco.editor.IStandaloneCodeEditor) => void,
+    onTextUpdate?: (editor: monaco.editor.IStandaloneCodeEditor) => void,
+}
+
+/**
+ * Wraps a `monaco-editor` instance created from the DOM rendered, which accepts a variety of props
```

Comment:
This docstring needs a bit of editing if we expect to reuse the component (e.g. to mention that it's read-only), but first what does "from the DOM rendered" mean?

Reply:
Right, I meant to say this component is designed solely to manage the states of a generic editor, without handling any log viewing states. This is in contrast to the MonacoInstance, which does involve manipulation of log states internally.